### PR TITLE
fix: copy prefix when showing it is disabled

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves address prefixes not copying

## How this PR fixes it

`showPrefix`/`copyPrefix` props have been added as per `safe-react` and the `<PrefixedEthHashInfo>` references settings to enable these accordingly.

## How to test it

Disable rendering of prefixes. Observe that the prefix is, however, copied.

### CODE REVIEW

File: src/components/common/EthHashInfo/index.tsx [hunk 1]
```diff
@@ -37,6 +42,7 @@ const EthHashInfo = ({
   hasExplorer,
 }: EthHashInfoProps): ReactElement => {
   const [fallbackToIdenticon, setFallbackToIdenticon] = useState(false)
+  const shouldPrefix = ethers.utils.isAddress(address)
```
Comment (DiogoSoaress) [inline src/components/common/EthHashInfo/index.tsx:45]:
```suggestion
  const isAddress = ethers.utils.isAddress(address)
```

File: src/components/common/EthHashInfo/index.tsx [hunk 2]
```diff
@@ -65,11 +71,13 @@ const EthHashInfo = ({
 
         <Box className={css.addressRow}>
           <Typography variant="body2" fontWeight="inherit" component="div" className={css.address}>
-            {prefix && <b>{prefix}:</b>}
-            {shortAddress ? shortenAddress(address) : address}
+            {showPrefix && shouldPrefix && prefix && <b>{prefix}:</b>}
+            {shortAddress && shortAddress ? shortenAddress(address) : address}
```
Comment (DiogoSoaress) [inline src/components/common/EthHashInfo/index.tsx:75]: Why this change?

File: src/components/common/EthHashInfo/index.tsx [hunk 3]
```diff
@@ -88,9 +96,16 @@ const PrefixedEthHashInfo = ({
...
 
   const name = showName ? props.name || addressBook[props.address] : undefined
-  const prefix = props.address.length === 42 && settings.shortName.show ? chain?.shortName : undefined
 
-  return <EthHashInfo prefix={prefix} {...props} name={name} />
+  return (
+    <EthHashInfo
+      prefix={chain?.shortName}
+      showPrefix={settings.shortName.show}
+      copyPrefix={settings.shortName.copy}
```
Comment (DiogoSoaress) [inline src/components/common/EthHashInfo/index.tsx:104]: We are passing these `settings` as props because eventually the `<EthHashInfo />` will come from a component library, right?

**Review by DiogoSoaress — APPROVED** [review]
👍 looks good. Couple of questions though.

**Comment by francovenica** [comment]
I see a bunch of questions still. I'll put it back in peer review. Send it to QA ready when merged

File: src/components/common/EthHashInfo/index.tsx (hunk 1 above)
Comment (iamacook) [reply src/components/common/EthHashInfo/index.tsx:45]: I would argue that this is more declarative as we don't want to prefix hashes. What do you think?

File: src/components/common/EthHashInfo/index.tsx (hunk 2 above)
Comment (iamacook) [reply src/components/common/EthHashInfo/index.tsx:75]: This was an oversight on my behalf. Good catch!

File: src/components/common/EthHashInfo/index.tsx (hunk 3 above)
Comment (iamacook) [reply src/components/common/EthHashInfo/index.tsx:104]: Exactly. I separated the local settings from the component.

File: src/components/common/EthHashInfo/index.tsx (hunk 1 above)
Comment (DiogoSoaress) [reply src/components/common/EthHashInfo/index.tsx:45]: I think it doesn't reflect our real intention in the logic. Perhaps `canPrefix` ? Once it can only prefix if the string is an address. Your call though

File: src/components/common/EthHashInfo/index.tsx (hunk 3 above)
Comment (DiogoSoaress) [reply src/components/common/EthHashInfo/index.tsx:104]: Super!

**Comment by francovenica** [comment]
Works fine.
